domain/schema/model.go

```diff
@@ -325,3 +341,99 @@ CREATE UNIQUE INDEX idx_unit_net_node
 ON unit (net_node_uuid);
 `)
 }
+
+func storageSchema() schema.Patch {
+	return schema.MakePatch(`
+CREATE TABLE storage_pool (
+    uuid TEXT PRIMARY KEY,
+    name TEXT NOT NULL,
+    -- TODO (manadart 2023-11-29): Storage pools in Mongo were stored as settings.
+    -- The types are sourced from the provider.
+    -- We can:
+    --   1. Leave it as text on the basis that we don't know up front what types the provider offers.
+    --   2. Make a lookup with all the known types, which would need maintenance over the long run.
+    --   3. Populate a lookup at model creation with the provider's known types.
+    -- Option 1 preserves the status quo.
+    type TEXT NOT NULL
+);
+
+CREATE TABLE storage_pool_attributes (
+    storage_pool_uuid TEXT NOT NULL,
+    key               TEXT,
+    value             TEXT,
+    CONSTRAINT       fk_storage_pool_attribute_pool
+        FOREIGN KEY  (storage_pool_uuid)
+        REFERENCES   storage_pool(uuid)
+);
+
+CREATE UNIQUE INDEX idx_storage_pool_attribute
+ON storage_pool_attributes (storage_pool_uuid, key);
+
+CREATE TABLE storage_type (
+    id   INT PRIMARY KEY,
+    type TEXT
+);
+
+CREATE UNIQUE INDEX idx_storage_type_type
+ON storage_type (type);
+
+INSERT INTO storage_type VALUES
```

Comment:
Can we make this "storage_kind".
Avoids the stutter and does match what we have in state/storage.go. We could over time look to update charm to use this terminology instead of StorageType like it does now.